### PR TITLE
CB-11126 ToC Generation Refactor

### PR DESCRIPTION
**Pull request by dblotsky**
**Note**: this PR is easier to review one commit at a time, and the commit messages roughly describe each change.

Changes:
- now `*-src.yml` files are mandatory for each docs version
- `*-src.yml` files get turned into `*-gen.yml` files
- renamed `*-manual.yml` files to the new `*-src.yml` format
- ToCs now contain only entries with `uri`s (unless they have children)
- pages now specify their ToC name using the `title` or `toc_title` front matter property
- removed the `name` property from ToC entries, moving it to the respective page where necessary

### CODE REVIEW

**Comment by dblotsky** [comment]
@riknoll @rakatyal @nikhilkh please review when you have a moment.

File: tools/bin/toc.js [hunk 1]
```diff
@@ -17,123 +17,57 @@
...
-                console.log(outputPath);
-                fs.writeFileSync(outputPath, tocText, 'utf8');
+            // read the input
+            fs.readFile(srcTocPath, function (error, data) {
+
+                // augment the ToC
+                var originalTocString  = data.toString();
+                var augmentedTocString = augment.augmentString(originalTocString, versionPath);
+                var warningComment     = console.log(util.generatedBy(__filename));
+                var output             = warningComment + "\n" + augmentedTocString;
```
Comment (riknoll) [inline tools/bin/toc.js]: This should just be the string and not console.log

Comment (dblotsky) [inline tools/bin/toc.js]: Good catch! Fixed.

**Comment by riknoll** [comment]
LGTM

**Pull request closed without merging**